Fix Makefile build target

sneal opened the pull request:
Add targets that download StemcellAutomation.zip dependencies from s3 to unblock local development of stembuild.

Since tar.exe, bosh-blobstore-s3.exe, bosh-blobstore-dav.exe [were removed from the BOSH agent repository](https://github.com/cloudfoundry/bosh-agent/commit/80d6b18efe25734d845884caf1d1c770ffc55551) this had the unfortunate side effect of breaking the stembuild make build target. The latest dependencies to build StemcellAutomation.zip are now downloaded from their respective s3 buckets that the official stembuild CI system uses.

The targets do require the use of `xq` (wrapper around jq that handles xml) to select the latest version of each dependencies in each s3 bucket.

rkoster reviewed (commented):
Could you also add a link to installation instructions for xq to the README somewhere around here: https://github.com/cloudfoundry/stembuild/blob/master/README.md#testing

aramprice commented:
Lint errors fixed on primary branch https://github.com/cloudfoundry/stembuild/commit/bbb4bb15d1e8938d5d687a1a00814185e6420600

sneal commented:
@rkoster I updated the local build instructions to mention `xq` and point to it's install docs. I also simplified the local build instructions (should be the same on Mac and Linux).

ystros commented on Makefile:
```diff
@@ -83,17 +86,19 @@ assets/local/bosh-agent.exe: $(BOSH_AGENT_SOURCES)
-	$(CP) $(BOSH_AGENT_REPO)/integration/windows/fixtures/bosh-blobstore-dav.exe assets/local
+assets/local/bosh-blobstore-dav.exe:
+	@echo "### Creating assets/local/bosh-blobstore-dav.exe"
+	$(eval BOSH_BLOBSTORE_DAV_FILE=$(shell curl -s $(BOSH_BLOBSTORE_DAV_URL) | xq -r '.ListBucketResult.Contents | last | .Key'))
```
What version of `xq` was in use here? Latest version (1.2.4) doesn't seem to have a `-r` option. The closest equivalent is `-x` for XPath directives, which doesn't seem to support jq-style piping used here.

As an aside, when `xq` either isn't installed or returns an error, the make process continues, simply zipping up the result of the content returned from `curl` as the EXE file.

sneal replied:
Well that's interesting. I will take a look at this again on March 12th - trying it on a clean Linux system.

sneal replied:
@ystros There are two different `xq` tools. I apparently have the one installed with [yq](https://github.com/kislyuk/yq) which really just wraps `jq`.

> yq also supports XML. The yq package installs an executable, xq, which [transcodes XML to JSON](https://www.xml.com/pub/a/2006/05/31/converting-between-xml-and-json.html) using [xmltodict](https://github.com/martinblech/xmltodict) and pipes it to jq.

Not confusing at all.

I will attempt to use the version of `xq` that you get when you explicitly install `xq` via `brew install xq` or `apt-get install xq`.

sneal replied:
Both `xq`s don't seem to be a good solution.

1. s3 api does not order the results in a useful way so just selecting the last element doesn't select the latest version.
1. s3 doesn't allow you to specify a server side sort
1. xpath is terrible, especially at sorting

Let me think about this. I'm leaning towards configurable env vars with defaults.

sneal replied:
I moved the sorting outside xq, so it works now. I also tried it on Linux.

sneal commented on README.md:
````diff
@@ -225,21 +225,25 @@ dep sync
+would like to build, for example:
 ```
-BOSH_AGENT_REPO=PathToBoshAgentRepo make STEMCELL_VERSION=StemcellVersion AUTOMATION_PATH=PathToStemcellAutomationZip build
+BOSH_AGENT_REPO=../bosh-agent STEMCELL_VERSION=2019.70 make build
````
`STEMBUILD_VERSION` which is an env var vs `STEMCELL_VERSION` which is a makefile var. I changed the instructions to use the env var, but it works the same way it used to.

rkoster commented:
Thanks! @sneal
